include/megaapi_impl.h

```diff
@@ -227,27 +227,66 @@ class MegaRecursiveOperation
     std::set<MegaTransferPrivate*> subTransfers;
     int mIncompleteTransfers = { 0 };
     MegaErrorPrivate mLastError = { API_OK };
+    LocalPath::separator_t mLocalSeparator;
+
+    // worker thread
+    std::thread mWorkerThread;
+
+    // thread id of main thread
+    std::thread::id mMainThreadId;
 };
 
+class TransferQueue;
 class MegaFolderUploadController : public MegaRequestListener, public MegaTransferListener, public MegaRecursiveOperation
```

Comment:
I think we can remove the inheritance from MegaRequestListener now - we no longer override any of its functions.